File: common_res/src/main/java/com/yyxnb/common_res/arouter/service/LoginService.java
package com.yyxnb.common_res.arouter.service;

import android.content.Context;

import com.alibaba.android.arouter.facade.template.IProvider;
import com.yyxnb.common_res.bean.UserBean;

/**
 * Login模块对外提供的所有功能
 */
public interface LoginService extends IProvider {

    UserBean getUserInfo();

    void updateUserInfo(UserBean userBean);

    void loginOut();

    boolean isLogin();

    @Deprecated
    void login(Context context);

}

File: common_base/src/main/java/com/yyxnb/common_base/arouter/service/LoginService.kt
package com.yyxnb.common_base.arouter.service

import android.content.Context
import com.alibaba.android.arouter.facade.template.IProvider
import com.yyxnb.common_base.bean.UserBean

/**
 * Login模块对外提供的所有功能
 */
interface LoginService : IProvider {

    val userInfo: UserBean?

    fun updateUserInfo(userBean: UserBean?)

    fun loginOut()

    val isLogin: Boolean

    @Deprecated("")
    fun login(context: Context?)
}